flang/lib/Semantics/check-omp-structure.h

```diff
@@ -70,6 +72,43 @@ class OmpStructureChecker
         ) {
   }
   using llvmOmpClause = const llvm::omp::Clause;
+  using ReductionModifier = parser::OmpReductionClause::ReductionModifier;
+  using Symbol = Fortran::semantics::Symbol;
+  class ScanReductionInfo {
+    std::set<Symbol *> usedInScan;
+    std::map<Symbol *, ReductionModifier> reductionMod;
```

Comment:
Do we still need this map and stack, now that we can go from the inclusive or exclusive clauses to the reduction symbol?

Reply:
I am trying to implement the alternative approach. The other alternative is as follows
-  to mark Symbol with `reduction Modifier` (using a flag or field)
- when scan directive is encountered, if `reductionmodifier` is not `inscan`, or flag is not `reduction`, error should be emitted.
- when loop directive is exited, for reduction variables, if flag is not `inclusive` or `exclusive` , error should be emitted. 
- Since reduction  declares a new symbol, and scan directive is closely nested within reduction, I believe stack is redundant for all valid programs (other semantic checks catches those errors)

I believe an extra flag should simplify the code.

Reply:
I was just asking a question. I have not thought through. If it needs further changes, then it is not needed for now. I will just have a think and reply tomorrow if that is OK.